ReactWindows/ChakraBridge/JsIndexedModulesUnbundle.cpp

```diff
@@ -0,0 +1,121 @@
+#include "pch.h"
+#include "ChakraHost.h"
+#include "JsIndexedModulesUnbundle.h"
+
+JsIndexedModulesUnbundle::JsIndexedModulesUnbundle(const wchar_t* sourcePath)
+{
+    this->sourcePath = sourcePath;
+}
+
+JsIndexedModulesUnbundle::~JsIndexedModulesUnbundle()
+{
+	if (fileHandle != NULL)
+	{
+		UnmapViewOfFile(byteBuffer);
+		CloseHandle(mapHandle);
+		CloseHandle(fileHandle);
+	}
+	else
+	{
+		delete[] byteBuffer;
+	}
+}
+
+JsModulesUnbundleModule* JsIndexedModulesUnbundle::GetModule(uint32_t index)
+{
+	ModuleData moduleData = moduleTable.modules[index];
+
+	// Create the source URL
+	auto filename = new wchar_t[_MAX_FNAME];
+	_itow_s(index, filename, _MAX_FNAME, 10);
+	wcscat_s(filename, _MAX_FNAME, L".js");
+
+	// Copy the script
+	auto script = new char[moduleData.length];
+	auto wcScript = new wchar_t[moduleData.length];
+	CopyMemory(script, byteBuffer + baseOffset + moduleData.offset, moduleData.length);
```

Comment:
Good suggestion.

---
In reply to: [103547834](https://github.com/ReactWindows/react-native-windows/pull/1017#discussion_r103547834) [](ancestors = 103547834)